sdk/ml/azure-ai-ml/azure/ai/ml/operations/_code_operations.py

```diff
@@ -192,6 +202,67 @@ def get(self, name: str, version: str) -> Code:
         :return: Code asset object.
         :rtype: ~azure.ai.ml.entities.Code
         """
+        return self._get(name=name, version=version)
+
+    # this is a public API but CodeOperations is hidden, so it may only monitor internal calls
+    @monitor_with_activity(logger, "Code.Download", ActivityType.PUBLICAPI)
+    def download(self, name: str, version: str, download_path: Union[PathLike, str]) -> None:
+        """Download content of a code.
+
+        :param str name: Name of the code.
+        :param str version: Version of the code.
+        :param Union[PathLike, str] download_path: Local path as download destination,
+            defaults to current working directory of the current user. Contents will be overwritten.
+        :raise: ResourceNotFoundError if can't find a code matching provided name.
+        """
+        output_dir = Path(download_path)
+        if output_dir.is_dir():
+            # an OSError will be raised if the directory is not empty
+            output_dir.rmdir()
+        output_dir.mkdir(parents=True)
+
+        code = self._get(name=name, version=version)
+
+        # TODO: how should we maintain this regex?
+        m = re.match(
+            r"https://(?P<account_name>.+)\.blob\.core\.windows\.net"
+            r"(:[0-9]+)?/(?P<container_name>.+)/(?P<blob_name>.*)",
+            code.path,
+        )
+        if not m:
+            raise ValueError(f"Invalid code path: {code.path}")
+
+        # 1. different content now saved in different container, so we need to override the container name
+        # 2. get credentials from datastore requires authorization to perform action
+        #    'Microsoft.MachineLearningServices/workspaces/datastores/listSecrets/action' over target datastore,
+        #    so we use local credential instead if so.
+        # check more information here:
+        # https://github.com/Azure/azureml_run_specification/blob/master/specs/create_workspace_asset_from_local_upload.md
+        try:
+            datastore_info = get_datastore_info(
+                self._datastore_operation,
+                # always use WORKSPACE_BLOB_STORE
+                name=_get_datastore_name(),
+                # local_credential=self._service_client._config.credential
+                container_name=m.group("container_name"),
+            )
+        except HttpResponseError:
+            datastore_info = get_datastore_info(
+                self._datastore_operation,
+                # always use WORKSPACE_BLOB_STORE
+                name=_get_datastore_name(),
+                local_credential=self._service_client._config.credential,
```

Comment:
shall this be the default path? As user should normally has permission to the default datastore?

Reply:
As Alex has said, current solution actually depends on their backend implementation of snapshots. Current default path is used by most artifact upload and all artifact download paths in DPv2 is more likely to be well informed and maintained if there are any changes.